[Wu Hanhui] ip

hanhuiice opened the pull request:
# Duke
> This is a quote.

Helps you remember things. It's

- text-based 
- ~~EASY~~ _SUPER_ EASY to learn 

What you need to do :clipboard:

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html)
2. add your tasks


Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main`
method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

hanhuiice committed:
Duke Level-1: Greet, Echo, Exit

hanhuiice committed:
Duke Level-2: Add, List

hanhuiice committed:
Duke Level-3: Mark as Done

hanhuiice committed:
Duke Level-4: ToDo, Event, Deadline

hanhuiice committed:
Duke A-TextUiTesting: Automated Text UI Testing

hanhuiice committed:
Duke Level-5: Handle Errors

hanhuiice committed:
Duke Level-6: Delete

hanhuiice committed:
Added javadoc comments.

hanhuiice committed:
Add storage function to save tasks

hanhuiice committed:
Add functionality to parse datetime from string

hanhuiice committed:
Merge branch 'branch-Level-7'

hanhuiice committed:
Merge branch 'branch-Level-8'

hanhuiice committed:
Update package structure

hanhuiice committed:
Extract out more classes

hanhuiice committed:
Update runtest.bat to use new project structure

hanhuiice committed:
Add find task function

hanhuiice committed:
Update to follow coding standard

hanhuiice committed:
Add javadoc to command classes

hanhuiice committed:
Merge branch 'branch-A-CodingStandard'

hanhuiice committed:
Merge branch 'branch-A-JavaDoc'

hanhuiice committed:
Merge branch 'branch-Level-9'

hanhuiice committed:
Add jUnit test for TaskList class

hanhuiice committed:
Add more javadoc to different classes

hanhuiice committed:
Add more junit tests

hanhuiice committed:
Merge remote-tracking branch 'origin/add-gradle-support'

hanhuiice committed:
Add gradle support

hanhuiice committed:
Add simple gui

hanhuiice committed:
Add checkstyle config file

limweiliang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,170 @@
+
+    /**
+     * Iteration 1:
+     * Creates a label with the specified text and adds it to the dialog container.
+     * @param text String containing text to add
```
Would fit the guidelines better to leave an empty line should be left between the description and parameter sections!

limweiliang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,170 @@
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
Was this meant to be rewritten?

limweiliang commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,86 @@
+ * Stores the user data in a text file.
+ */
+public class Storage {
+    private final File directory = new File("data");
+    private final File file = new File("data/tasks.txt");
```
Would it be better to name these in ALL_CAPS and make them static variables since they appear to be constants?

limweiliang commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,86 @@
+        }
+    }
+
+    private Task stringToTask(String task) {
```
Maybe this could be a verb to better suit the coding standards? Might I suggest `convertStringToTask`?

limweiliang commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+ * Represents a task that has deadline.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
```
I think it might suit the coding standards better if this was a noun! e.g. `dueDate`

limweiliang reviewed (commented):
LGTM! 👍 The code was quite easy to read and understand.

However, I do have some minor coding style suggestions!

hanhuiice committed:
Update duke response for gui

DALIN-Prog commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+     * @return The string description.
+     */
+    public String taskListToString() {
+        StringBuilder result = new StringBuilder();
```
Great attempt in speeding up string concatenation!

DALIN-Prog commented on src/main/java/duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
```
Probably you have forgotten your javadocs

DALIN-Prog commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,122 @@
+     * @return The command corresponding to the input.
+     * @throws DukeException If the input string have errors in its format.
+     */
+    public static Command parse(String input) throws DukeException {
```
I like how you have created separate functions for each case to keep your method short!

Good understanding of switch case is displayed here (i.e no unnecessary break)

DALIN-Prog commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,122 @@
+    private static Command extractDeadline(String[] details) {
+        try {
+            String[] args = details[1].split(" /by ");
+            LocalDateTime by = LocalDateTime.parse(args[1], DateTimeFormatter.ofPattern("d/M/y HHmm"));
```
Perhaps instead of by, you can consider using dateTime

DALIN-Prog commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,86 @@
+     *
+     * @return The tasks stored in an array list.
+     */
+    public ArrayList<Task> retrieveTaskList() {
```
Perhaps to better match the name of the variable you return, your method name can be retrieveTasks

DALIN-Prog reviewed (commented):
Overall I have found your code easy to read. Great work!

hanhuiice committed:
Update dialogbox gui layout

hanhuiice committed:
Remove parts of text ui and its relevant methods

hanhuiice committed:
Improve code quality

hanhuiice committed:
Merge pull request #1 from hanhuiice/branch-A-CodeQuality

Improve code quality

hanhuiice committed:
Add java assert checks

hanhuiice committed:
Merge pull request #2 from hanhuiice/branch-A-Assertions

Add java assert checks

hanhuiice committed:
Remove ui class

Ui class: Remove the class.

Command execution results can be passed directly to the gui, text ui
is not needed anymore.

Delete ui class and update command classes to stop taking in ui as
parameter instead they will return execution result strings to be
passed to gui.

hanhuiice committed:
Add message class

Message class: extract a message class

The error messages are placed at one location which provides easy
modification to them when necessary in the future.

Extract error message from different methods in parser class and put
them under message class.

hanhuiice committed:
Improve code quality

hanhuiice committed:
Add reminders to be printed when duke starts

Duke will get a list of upcoming tasks within next two days when
initialize and get gui to print reminder after welcome message.

The behaviour gives user easy access to upcoming task without
requiring any actions from them.

hanhuiice committed:
Set up CI

hanhuiice committed:
Improve code quality

hanhuiice committed:
Update user guide

hanhuiice committed:
Update user guide
